packages/lab/src/avatar/Avatar.tsx

```diff
@@ -1,74 +1,88 @@
-import { makePrefixer, useId } from "@salt-ds/core";
+import { makePrefixer } from "@salt-ds/core";
+import { UserIcon } from "@salt-ds/icons";
 import { clsx } from "clsx";
-import { forwardRef, HTMLAttributes, ImgHTMLAttributes } from "react";
-import { classBase } from "./internal/constants";
-import { DefaultAvatar } from "./internal/DefaultAvatar";
+import { forwardRef, HTMLAttributes, ReactNode } from "react";
 import { useLoaded } from "./internal/useLoaded";
 
 import "./Avatar.css";
 
+export type InitialsGetter = (name?: string) => string;
+
 export interface AvatarProps extends HTMLAttributes<HTMLDivElement> {
-  alt?: string;
-  id?: string;
-  imgProps?: ImgHTMLAttributes<HTMLImageElement>;
-  size?: "small" | "medium" | "large";
-  sizes?: string;
+  /**
+   * The name that Avatar represents.
+   */
+  name?: string;
+  /**
+   * Defines the function that gets initials. Default is capital first letter of each separate word in name.
+   * If a function is not passed or returns undefined, Avatar will default to Icon.
+   */
+  initialsGetter?: InitialsGetter;
+  /**
+   * Image src of Avatar.
+   */
   src?: string;
-  srcSet?: string;
-  title?: string;
+  /**
+   * Multiplier for the base avatar.
+   */
+  size?: number;
+  /**
+   * Icon to be used as a default item. Defaults to `UserIcon`
+   */
+  fallbackIcon?: ReactNode;
 }
 
-const withBaseName = makePrefixer(classBase);
+const withBaseName = makePrefixer("saltAvatar");
+const DEFAULT_AVATAR_SIZE = 2; // medium
+
+const initialsDefaultGetter = (name?: string) =>
+  name
+    ?.split(" ")
+    .slice(0, 2)
```

Comment:
Still unsure about this limit

Reply:
We need to retain the 2 character limit with the new component. Although it would be nice to offer more it adds complexity. If the characters are, for instance WMW, they'd be too wide for the avatar and would either overlap the edge or have to wrap. Alternatively we'd have to look into ways the font size could be reduced to accommodate 3+ characters which would be further work. 

This isn't to say we shouldn't look into it in the future, as a person with a double-barrelled surname I can see them value in doing it, but for now I don't think its a priority.